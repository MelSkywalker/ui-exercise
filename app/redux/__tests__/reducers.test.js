import React from 'react';
import emailsReducer, { LOAD_EMAIL_LIST } from '../../redux/reducers';

describe('Redux | reducers', () => {
	const payload = {
		messages: [
			{
				"id": "1",
				"subject": "Hello",
				"tags": ["tag02"]
			},
			{
				"id": "2",
				"subject": "hello!!!",
				"tags": ["tag01", "tag02"]
			}
		]
	}

	it('should return the initial state', () => {
		expect(emailsReducer(undefined, {})).toMatchSnapshot();
	});

	it('should populate the state with emails and tags', () => {
		const state = emailsReducer(undefined, {
			type: LOAD_EMAIL_LIST,
			payload
		});
		expect(state.emailsById['1'].subject).toEqual('Hello');
		expect(state.emailsIdArray).toEqual(['1', '2']);
		expect(state.tags).toHaveLength(2); // check it is not repeating tags
		expect(state).toMatchSnapshot();
	});
})