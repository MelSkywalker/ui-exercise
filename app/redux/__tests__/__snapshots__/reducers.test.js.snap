// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux | reducers should populate the state with emails and tags 1`] = `
Object {
  "emailsById": Object {
    "1": Object {
      "id": "1",
      "subject": "Hello",
      "tags": Array [
        "travel",
      ],
    },
    "2": Object {
      "id": "2",
      "subject": "hello!!!",
      "tags": Array [
        "work",
        "travel",
      ],
    },
  },
  "emailsIdArray": Object {
    "all": Array [
      "1",
      "2",
    ],
    "travel": Array [
      "1",
      "2",
    ],
    "work": Array [
      "2",
    ],
  },
  "filter": "all",
  "messageCount": 2,
  "selectedEmails": Array [],
  "tags": Array [
    "travel",
    "work",
  ],
}
`;
